crates/db_schema/src/source/community_report.rs

```diff
@@ -0,0 +1,60 @@
+use crate::newtypes::{CommunityId, CommunityReportId, DbUrl, PersonId};
+#[cfg(feature = "full")]
+use crate::schema::community_report;
+use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+#[cfg(feature = "full")]
+use ts_rs::TS;
+
+#[skip_serializing_none]
+#[derive(PartialEq, Eq, Serialize, Deserialize, Debug, Clone)]
+#[cfg_attr(
+  feature = "full",
+  derive(Queryable, Selectable, Associations, Identifiable, TS)
+)]
+#[cfg_attr(
+  feature = "full",
+  diesel(belongs_to(crate::source::community::Community))
+)]
+#[cfg_attr(feature = "full", diesel(table_name = community_report))]
+#[cfg_attr(feature = "full", diesel(check_for_backend(diesel::pg::Pg)))]
+#[cfg_attr(feature = "full", ts(export))]
+/// A comment report.
+pub struct CommunityReport {
+  pub id: CommunityReportId,
+  pub creator_id: PersonId,
+  pub community_id: CommunityId,
+  pub original_community_name: String,
+  pub original_community_title: String,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub original_community_description: Option<String>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub original_community_sidebar: Option<String>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub original_community_icon: Option<String>,
+  #[cfg_attr(feature = "full", ts(optional))]
+  pub original_community_banner: Option<String>,
```

Comment:
Not sure if the icon and banner are totally necessary, but no reason not to have them I spose.

Reply:
If I understand correctly, reports store the original content so the creator of the reported thing can't edit it to remove evidence of the offense. It's possible that a community's icon or banner is the reason for reporting, just like the url of a post.